src/common/SurgeStorage.cpp

```diff
@@ -356,8 +356,10 @@ void SurgeStorage::load_wt_wt(string filename, Wavetable* wt)
    wt_header wh;
    memset(&wh, 0, sizeof(wt_header));
 
-   fread(&wh, sizeof(wt_header), 1, f);
-   if (wh.tag != vt_read_int32BE('vawt'))
+   if (fread(&wh, sizeof(wt_header), 1, f) != 1)
+      fprintf(stderr, "SurgeStorage: reading the wavetable header failed.\n");
```

Comment:
Maybe it needs to further deal with the error here? Could it return an error code or throw an exception? Sorry I'm not too familiar with that code myself.

Reply:
I would suggest at this point a log message and a `FIXME` comment to implement proper error handling later. At least now we are aware if this happens (which should be rarely). It fixes the compiler warning and makes the error visible.

Things like exception handling and RTTI should be decided in the project level. For example, **WebKit** uses neither for maximum portability. Personally I think they are good for UI and application level code and not that good for engine level code but I don't think this is right time to nail that. For me here my personal preferences are less important than project wide conventions.